<?xml version="1.0"?>
<launch>
    <rosparam command="load" file="$(find point_cloud_object_detector)/config/color_param.yaml"/>
    <arg name="manager_name" default="nodelet_magager" />
    <node pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager" output="screen"/>
    <!--robot1-->
    <!-- <node pkg="point_cloud_object_detector" type="point_cloud_object_detector_node" name="object_detector1" output="screen" > -->
    <!--     <param name="tolerance" value="0.2" type="double"/> -->
    <!--     <param name="min_cluster_size" value="500" type="int" /> -->
    <!--     <param name="max_cluster_size" value="6000" type="int" /> -->
    <!--     <param name="color_cluster_th" value="40" type="int" /> -->
    <!--     <rosparam command="load" file="$(find point_cloud_object_detector)/config/color_param.yaml"/> -->
    <!--     <remap from="/object_positions" to="robot1/object_positions" /> -->
    <!--     <remap from="/darknet_ros/bounding_boxes" to="robot1/darknet_ros/bounding_boxes" /> -->
    <!--     <remap from="/camera/depth_registered/points" to="robot1/camera/depth_registered/points" /> -->
    <!-- </node> -->
    <!-- robot2 -->
    <node pkg="point_cloud_object_detector" type="point_cloud_object_detector_node" name="object_detector2" output="screen" >
        <param name="tolerance" value="0.2" type="double"/>
        <param name="min_cluster_size" value="500" type="int" />
        <param name="max_cluster_size" value="6000" type="int" />
        <param name="color_cluster_th" value="40" type="int" />
        <param name="img_th" value="40" type="int" />
        <param name="param_file_name" value="/home/amsl/catkin_ws/src/point_cloud_object_detector/config/color_param.yaml" type="string" />
        <remap from="/object_positions" to="robot2/object_positions" />
        <remap from="/darknet_ros/bounding_boxes" to="robot2/darknet_ros/bounding_boxes" />
        <remap from="/camera/depth_registered/points" to="robot2/camera/depth_registered/points" />
        <remap from="/camera/image_rect_color/compressed" to="robot2/camera/color/image_rect_color/compressed" />
        <remap from="/camera/image_rect_color" to="robot2/camera/color/image_rect_color" />
    </node>
    <!-- <include file="$(find point_cloud_object_detector)/launch/rosbag_recorder.launch"> -->
    <!--     <arg name="robot" value="6"/> -->
    <!-- </include> -->
    <!--robot3-->
    <node pkg="point_cloud_object_detector" type="point_cloud_object_detector_node" name="object_detector3" output="screen" >
        <param name="tolerance" value="0.2" type="double"/>
        <param name="min_cluster_size" value="500" type="int" />
        <param name="max_cluster_size" value="6000" type="int" />
        <param name="color_cluster_th" value="40" type="int" />
        <param name="img_th" value="40" type="int" />
        <param name="param_file_name" value="/home/amsl/catkin_ws/src/point_cloud_object_detector/config/color_param.yaml" type="string" />
        <remap from="/object_positions" to="robot3/object_positions" />
        <remap from="/darknet_ros/bounding_boxes" to="robot3/darknet_ros/bounding_boxes" />
        <remap from="/camera/depth_registered/points" to="robot3/camera/depth_registered/points" />
        <remap from="/camera/image_rect_color/compressed" to="robot3/camera/color/image_rect_color/compressed" />
        <!-- <remap from="/camera/image_rect_color" to="robot3/camera/color/image_rect_color" /> -->
    </node>
    <!-- <include file="$(find point_cloud_object_detector)/launch/rosbag_recorder.launch"> -->
    <!--     <arg name="robot" value="3"/> -->
    <!-- </include> -->
    <!--robot4-->
    <!-- <node pkg="point_cloud_object_detector" type="point_cloud_object_detector_node" name="object_detector4" output="screen" > -->
    <!--     <param name="tolerance" value="0.2" type="double"/> -->
    <!--     <param name="min_cluster_size" value="500" type="int" /> -->
    <!--     <param name="max_cluster_size" value="6000" type="int" /> -->
    <!--     <param name="color_cluster_th" value="40" type="int" /> -->
    <!--     <rosparam command="load" file="$(find point_cloud_object_detector)/config/color_param.yaml"/> -->
    <!--     <remap from="/object_positions" to="robot4/object_positions" /> -->
    <!--     <remap from="/darknet_ros/bounding_boxes" to="robot4/darknet_ros/bounding_boxes" /> -->
    <!--     <remap from="/camera/depth_registered/points" to="robot4/camera/depth_registered/points" /> -->
    <!--     <remap from="/camera/image_rect_color/compressed" to="robot4/camera/color/image_rect_color/compressed" /> -->
    <!-- </node> -->
    <!-- <include file="$(find point_cloud_object_detector)/launch/rosbag_recorder.launch"> -->
    <!--     <arg name="robot" value="4"/> -->
    <!-- </include> -->
    <!--robot5-->
    <node pkg="point_cloud_object_detector" type="point_cloud_object_detector_node" name="object_detector5" output="screen" >
        <param name="tolerance" value="0.2" type="double"/>
        <param name="min_cluster_size" value="500" type="int" />
        <param name="max_cluster_size" value="6000" type="int" />
        <param name="color_cluster_th" value="40" type="int" />
        <param name="img_th" value="40" type="int" />
        <rosparam command="load" file="$(find point_cloud_object_detector)/config/color_param.yaml"/>
        <remap from="/object_positions" to="robot5/object_positions" />
        <remap from="/darknet_ros/bounding_boxes" to="robot5/darknet_ros/bounding_boxes" />
        <remap from="/camera/depth_registered/points" to="robot5/camera/depth_registered/points" />
        <remap from="/camera/image_rect_color" to="robot5/camera/color/image_rect_color" />
    </node>
    <!-- <include file="$(find point_cloud_object_detector)/launch/rosbag_recorder.launch"> -->
    <!--     <arg name="robot" value="5"/> -->
    <!-- </include> -->
    <!--robot6-->
    <node pkg="point_cloud_object_detector" type="point_cloud_object_detector_node" name="object_detector6" output="screen" >
        <param name="tolerance" value="0.2" type="double"/>
        <param name="min_cluster_size" value="500" type="int" />
        <param name="max_cluster_size" value="6000" type="int" />
        <param name="color_cluster_th" value="40" type="int" />
        <param name="img_th" value="40" type="int" />
        <param name="param_file_name" value="/home/amsl/catkin_ws/src/point_cloud_object_detector/config/color_param.yaml" type="string" />
        <remap from="/object_positions" to="robot6/object_positions" />
        <remap from="/darknet_ros/bounding_boxes" to="robot6/darknet_ros/bounding_boxes" />
        <remap from="/camera/depth_registered/points" to="robot6/camera/depth_registered/points" />
        <!-- <remap from="/camera/image_rect_color/compressed" to="robot6/camera/color/image_rect_color/compressed" /> -->
        <remap from="/camera/image_rect_color" to="robot6/camera/color/image_rect_color" />
    </node>
    <node pkg="nodelet" type="nodelet" name="point_cloud_xyzrgb6" args="load depth_image_proc/point_cloud_xyzrgb $(arg manager_name)">
        <remap from="rgb/camera_info" to="robot6/camera/color/camera_info" />
        <remap from="rgb/image_rect_color" to="robot6/camera/color/image_rect_color" />
        <remap from="depth_registered/image_rect" to="robot6/camera/aligned_depth_to_color/image_raw" />
        <remap from="depth_registered/points" to="robot6/camera/depth_registered/points" />
        <!-- <remap from="rgb/image_rect_color" to="robot6/camera/color/image_rect_color/compressed" /> -->
        <!-- <remap from="rgb/image_rect" to="robot6/camera/aligned_depth_to_color/image_raw/compressed" /> -->
        <!-- <remap from="points" to="robot6/camera/depth_registered/points" /> -->
    </node>
    <!-- <include file="$(find point_cloud_object_detector)/launch/rosbag_recorder.launch"> -->
    <!--     <arg name="robot" value="6"/> -->
    <!-- </include> -->
    <!--  -->
    <!--make single topic-->
    <!-- <node pkg="camera_rename" type="make_single_topic" name="make_single_topic" /> -->
    <!-- <node pkg="camera_rename" type="make_single_topic_comp" name="make_single_topic_comp" /> -->

    <!-- <include file="/home/amsl/catkin_ws/src/darknet_ros/darknet_ros/launch/wrs.launch" /> -->
    <include file="/home/amsl/catkin_ws/src/darknet_ros/darknet_ros/launch/use_cmp_img.launch" />

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find point_cloud_object_detector)/config/realtime_locali.rviz"/>


    <!-- <include file="/home/amsl/catkin_ws/src/distance_based_localizer/launch/norm_robot.launch"> -->
    <!--     <arg name="robot" value="robot5"/> -->
    <!-- </include> -->
    <!-- <include file="/home/amsl/catkin_ws/src/distance_based_localizer/launch/norm_robot.launch"> -->
    <!--     <arg name="robot" value="robot6"/> -->
    <!-- </include> -->
</launch>
